generate_file_section.py

```diff
@@ -0,0 +1,290 @@
+import argparse
+import csv
+import fnmatch
+import os
+import re
+from pathlib import Path
+from pprint import pformat
+from warnings import warn
+
+
+def delete_commonpath(longer_path, prefix):
+    """return string with deleted common path."""
+
+    common_path = os.path.commonpath((longer_path, prefix)).split('/')
+    return '/'.join(longer_path.split('/')[len(common_path):])
+
+
+def locate_record(root, python3_sitelib, python3_sitearch):
+    """return path to record stripped of root path."""
+
+    record_path = list((Path(root) / Path(python3_sitelib[1:])).glob('*.dist-info/RECORD'))
+    record_path.extend(list((Path(root) / Path(python3_sitearch[1:])).glob('*.dist-info/RECORD')))
+
+    if len(record_path) == 0:
+        raise FileNotFoundError("Did not find RECORD file")
+    if len(record_path) > 1:
+        raise FileExistsError("Multiple *.dist-info directories")
+
+    record_path = str(record_path[0])
+    return "/" + delete_commonpath(record_path, root)
+
+
+def read_record(root, record_path):
+    """return parsed list [[[path], [hash], [size]], ...]"""
+
+    with open(f"{root}/{record_path}", newline='') as f:
+        content = csv.reader(f, delimiter=',', quotechar='"', lineterminator=os.linesep)
+        return list(content)
+
+
+def parse_record(record_path, record_content):
+    """return list of paths stripped of root
+
+    params:
+    record_path: path to record file stripped of root
+    record_content: list of files relative to directory where dist-info directory is
+
+    Example:
+        parse_record("/usr/lib/python3.7/site-packages/requests-2.22.0.dist-info/RECORD", ["requests", ...])
+            -> ["/usr/lib/python3.7/site-packages/requests", ...]
+    """
+
+    record_path = "/".join(record_path.split('/')[:-2])
+    files = [os.path.normpath(os.path.join(record_path, row[0])) for row in record_content]
+    return files
+
+
+def pattern_filter(pattern, parsed_record_content):
+    """filter list by given pattern."""
+
+    matched = []
+    comp = re.compile(pattern)
+    for path in parsed_record_content:
+        if comp.search(path):
+            matched.append(path)
+    return matched
+
+
+def find_metadata(parsed_record_content, python3_sitedir, record_path):
+    """go through parsed RECORD content, returns tuple:
+    (path to directory containing metadata, [paths to all metadata files])."""
+
+    dist_info = re.search(f"{re.escape(python3_sitedir)}/[^/]*", record_path)[0] + "/"
+
+    return dist_info, [*pattern_filter(f"{re.escape(dist_info)}.*", parsed_record_content)]
+
+
+def find_extension(python3_sitedir, parsed_record_content):
+    """list paths to extensions"""
+
+    return pattern_filter(f"{re.escape(python3_sitedir)}/[^/]*\\.so$", parsed_record_content)
```

Comment:
Use the Path.match method instead, regular expression are overkill:

```suggestion
    return [path for path in parsed_record_content
            if path.parent == python3_sitedir and path.match('*.so')]
```